diff -Naur a/drivers/net/wireless/marvell/Kconfig b/drivers/net/wireless/marvell/Kconfig
--- a/drivers/net/wireless/marvell/Kconfig	2017-09-17 01:47:51.000000000 +0300
+++ b/drivers/net/wireless/marvell/Kconfig	2017-09-22 10:15:55.000000000 +0300
@@ -14,6 +14,7 @@
 source "drivers/net/wireless/marvell/libertas/Kconfig"
 source "drivers/net/wireless/marvell/libertas_tf/Kconfig"
 source "drivers/net/wireless/marvell/mwifiex/Kconfig"
+source "drivers/net/wireless/marvell/mwlwifi/Kconfig"
 
 config MWL8K
 	tristate "Marvell 88W8xxx PCI/PCIe Wireless support"
diff -Naur a/drivers/net/wireless/marvell/Makefile b/drivers/net/wireless/marvell/Makefile
--- a/drivers/net/wireless/marvell/Makefile	2017-09-17 01:47:51.000000000 +0300
+++ b/drivers/net/wireless/marvell/Makefile	2017-09-22 10:19:39.000000000 +0300
@@ -4,3 +4,5 @@
 obj-$(CONFIG_MWIFIEX)	+= mwifiex/
 
 obj-$(CONFIG_MWL8K)	+= mwl8k.o
+
+obj-$(CONFIG_MWLWIFI)		+= mwlwifi/
